### PR TITLE
Refactor tm_parser_format_function: merge double `switch (lang)`

### PR DESCRIPTION
**Pull request by ntrel**

### CODE REVIEW

**Comment by elextr** [comment]
Hmmm, yes the original didn't make sense did it?

File: src/tagmanager/tm_parser.c [hunk 1]
```diff
@@ -1531,38 +1531,32 @@ gchar *tm_parser_format_function(TMParserType lang, const gchar *fname, const gc
...
 				/* retval before function */
 				g_string_prepend_c(str, ' ');
 				g_string_prepend(str, retval);
 				break;
 		}
+		if (sep)
+		{
+			g_string_append(str, sep);
+			g_string_append(str, retval);
+		}
```
Comment (b4n) [inline src/tagmanager/tm_parser.c:1559]: I'd rather have an if/else here if the value of `sep` is the criteria, like that I find the code weird as well.
```suggestion
				break;
		}

		if (sep)
		{
			/* retval after function */
			g_string_append(str, sep);
			g_string_append(str, retval);
		}
		else
		{
			/* retval before function */
			g_string_prepend_c(str, ' ');
			g_string_prepend(str, retval);
		}
```

**Comment by b4n** [comment]
> Hmmm, yes the original didn't make sense did it?

It did (see how the outer switch caught all the inner one's values), it was just a bit of a bizarre way of laying out the code.

**Comment by elextr** [comment]
> > Hmmm, yes the original didn't make sense did it?

>  it was just a bit of a bizarre way of laying out the code.

See edit adding the words I thought would be understood. :grin:

**Pull request merged**